packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -659,6 +663,10 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  const numSelectedRows =
+    selectionMode === 'single'
+      ? 0
+      : rows.filter((row) => row.isSelected).length;
```

Comment:
comment from @rohan-kadkol 

> Are we also counting subrows in this `numSelectedRows` count?
> 
> If so, then you might need to use `flatRows`. It is an ordered array of all rows (even subrows).

Reply:
This should also fix a bug where you apply filter and it no longer shows how many rows are selected if they are not visible.

Reply:
This has been changed to use `flatRows` here, but now the row selection count includes all rows and their subrows.

Imagine a table with Row 1 which has subrows Row 1a and Row 1b. If Row 1 is selected, the row count shows as 3 rows selected (Row 1, Row 1a, and Row 1b). If all of a row's subrows are selected, that row is also counted as selected. Is this an issue?

Reply:
I think that is how it works. If one row is selected, all subrows, and subrows' subrows (excl. disabled) have to be selected